Sources/KSCrashDemangleFilter/KSCrashReportFilterDemangle.m

```diff
@@ -0,0 +1,168 @@
+    char *demangled = ksdm_demangleSwift(symbol.UTF8String);
+    if (demangled != NULL) {
+        NSString *result = [[NSString alloc] initWithBytesNoCopy:demangled
+                                                          length:strlen(demangled)
+                                                        encoding:NSUTF8StringEncoding
+                                                    freeWhenDone:YES];
+        KSLOG_DEBUG(@"Demangled a Swift symbol '%@' -> '%@'", symbol, result);
+        return result;
+    }
+#endif
+    return nil;
+}
+
++ (NSString *)demangledSymbol:(NSString *)symbol
+{
+    return [self demangledCppSymbol:symbol] ?: [self demangledSwiftSymbol:symbol];
+}
+
+/** Recurcively demangles strings within the report.
+ * @param reportObj An object within the report (dictionary, array, string etc)
+ * @param path An array of strings representing keys in dictionaries. An empty key means an itteration within the array.
+ * @param depth Current depth of the path
+ * @return An updated object or `nil` if no changes were applied.
+ */
++ (id)demangleReportObj:(id)reportObj path:(NSArray<NSString *> *)path depth:(NSUInteger)depth
+{
+    // Check for NSString and try demangle
+    if (depth == path.count) {
+        if ([reportObj isKindOfClass:[NSString class]] == NO) {
+            return nil;
+        }
+        NSString *demangled = [self demangledSymbol:reportObj];
+        return demangled;
+    }
+
+    NSString *pathComponent = path[depth];
+
+    // NSArray:
+    if (pathComponent.length == 0) {
+        if ([reportObj isKindOfClass:[NSArray class]] == NO) {
+            return nil;
+        }
+        NSArray *reportArray = reportObj;
+        NSMutableArray *__block result = nil;
+        [reportArray enumerateObjectsUsingBlock:^(id obj, NSUInteger idx, BOOL *_Nonnull stop) {
+            id demangled = [self demangleReportObj:obj path:path depth:depth + 1];
+            if (demangled != nil && result == nil) {
+                // Initializing the updated array only on first demangled result
+                result = [NSMutableArray arrayWithCapacity:reportArray.count];
+                for (NSUInteger subIdx = 0; subIdx < idx; ++subIdx) {
+                    [result addObject:reportArray[subIdx]];
+                }
+            }
+            result[idx] = demangled ?: obj;
+        }];
+        return [result copy];
+    }
+
+    // NSDictionary:
+    if ([reportObj isKindOfClass:[NSDictionary class]] == NO) {
+        return nil;
+    }
+    NSDictionary *reportDict = reportObj;
+    id demangledElement = [self demangleReportObj:reportDict[pathComponent] path:path depth:depth + 1];
+    if (demangledElement == nil) {
+        return nil;
+    }
+    NSMutableDictionary *result = [reportDict mutableCopy];
+    result[pathComponent] = demangledElement;
+    return [result copy];
+}
+
+- (void)filterReports:(NSArray<id<KSCrashReport>> *)reports onCompletion:(KSCrashReportFilterCompletion)onCompletion
```

Comment:
I think marking which code refers to the `KSCrashReportFilter` protocol would make the code easier to understand.